src/encryptsave/mod.rs

```diff
@@ -0,0 +1,175 @@
+mod ll;
+pub mod errors;
+
+
+macro_rules! tox_res {
+    ( $res:ident <- $rexpr:expr, $err:ident, $r:expr ) => {
```

Comment:
You don't need it. https://github.com/suhr/rstox/blob/master/src/core/mod.rs#L348 introduces ident only because expr use it (for example, `&mut err` in https://github.com/suhr/rstox/blob/master/src/core/mod.rs#L478).
You can use just `let mut out = $rexpr;` since macros are hygienic.
Also consider something like `($err:ident, $e:expr, if $cond:expr)`